controllers/controllers_suite_test.go

```diff
@@ -1,29 +1,76 @@
-package serviceaccount_test
+package controllers_test
 
 import (
 	"context"
 	"crypto/rand"
 	"fmt"
+	"go/build"
 	"math/big"
+	"net"
+	"net/url"
+	"path/filepath"
 	"strconv"
 	"testing"
 
+	gkehub "cloud.google.com/go/gkehub/apiv1beta1"
 	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"google.golang.org/api/option"
+	gkehubpb "google.golang.org/genproto/googleapis/cloud/gkehub/v1beta1"
+	"google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/client-go/kubernetes/scheme"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	kcapi "k8s.io/client-go/tools/clientcmd/api"
+	infra "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	"github.com/giantswarm/workload-identity-operator-gcp/controllers"
 	serviceaccount "github.com/giantswarm/workload-identity-operator-gcp/controllers"
+	"github.com/giantswarm/workload-identity-operator-gcp/pkg/gke"
 	"github.com/giantswarm/workload-identity-operator-gcp/tests"
 	//+kubebuilder:scaffold:imports
 )
 
+type FakeGKEServer struct {
```

Comment:
Similarly to [what we do in other operators](https://github.com/giantswarm/capg-firewall-rule-operator/blob/main/pkg/firewall/client.go), we should create our own client that wraps the google one. Then [we can fake our own client](https://github.com/giantswarm/capg-firewall-rule-operator/blob/main/controllers/gcpcluster_test.go#L54) in our tests, and [create a new integration test for our client](https://github.com/giantswarm/capg-firewall-rule-operator/blob/main/tests/integration/firewall/client_test.go) that tests that the client works using real google API.